contracts/test/TestV1Allocation.sol

```diff
@@ -0,0 +1,141 @@
+      global: JBGlobalFundingCycleMetadata({allowSetTerminals: false, allowSetController: false}),
+      reservedRate: 5000, //50%
+      redemptionRate: 5000, //50%
+      ballotRedemptionRate: 0,
+      pausePay: false,
+      pauseDistributions: false,
+      pauseRedeem: false,
+      pauseBurn: false,
+      allowMinting: false,
+      allowChangeToken: false,
+      allowTerminalMigration: false,
+      allowControllerMigration: false,
+      holdFees: false,
+      useTotalOverflowForRedemptions: false,
+      useDataSourceForPay: false,
+      useDataSourceForRedeem: false,
+      dataSource: address(0)
+    });
+
+    _terminals.push(jbETHPaymentTerminal());
+
+    _fundAccessConstraint.push(JBFundAccessConstraints({
+        terminal: jbETHPaymentTerminal(),
+        token: jbLibraries().ETHToken(),
+        distributionLimit: 2 ether,
+        overflowAllowance: 0,
+        distributionLimitCurrency: jbLibraries().ETH(),
+        overflowAllowanceCurrency: jbLibraries().ETH()
+    }));
+
+    JBGroupedSplits[] memory _groupedSplits = new JBGroupedSplits[](1); // Default empty
+    
+    // mimicing a v3 project launch since there was dependency issue with importing both v2 & v3 contract togethet since the almost every contract is the same
+    evm.prank(_allocationDDestinationProjectOwner);
+    _allocationDestinationProjectId = controller.launchProjectFor(
+      _allocationDDestinationProjectOwner,
+      _projectMetadata,
+      _data,
+      _metadata,
+      block.timestamp,
+      _groupedSplits,
+      _fundAccessConstraint,
+      _terminals,
+      ''
+    );
+
+    PayoutMod[] memory _payOutMods = new PayoutMod[](1);
+    _payOutMods[0] = PayoutMod({
+        preferUnstaked: false,
+        percent: 10000,
+        lockedUntil: 0,
+        beneficiary: _allocationDDestinationProjectOwner,
+        allocator: _v1Allocator,
+        projectId: uint56(_allocationDestinationProjectId)
+    });
+
+    evm.prank(_allocationSourceProjectOwner);
+    _modStore.setPayoutMods(_allocationSourceProjectId, block.timestamp, _payOutMods);
+  }
+
+  function test_V1Allocation() public {
+    JBETHPaymentTerminal _v3Terminal = jbETHPaymentTerminal();
+    TerminalV1_1 _v1Terminal = terminal();
+    address _user = address(bytes20(keccak256('user')));
+
+    // fund user
+    evm.deal(_user, 1 ether);
+    
+    // pay project
+    evm.prank(_user);
+    _v1Terminal.pay{value: 1 ether}(
+      _allocationSourceProjectId,
+      _user,
+      'Take my money',
+       false
+    );
+
+    assertEq(_v1Terminal.balanceOf(_allocationSourceProjectId), 1 ether);
+    
+    // distribute thhe funds to another project
+    evm.prank(_allocationSourceProjectOwner);
+    _v1Terminal.tap(
+      _allocationSourceProjectId,
+      1 ether,
+      0, // Currency
+      0 // Min wei out
+    );
+    
+    // considering the fee
+    assert(
+      jbPaymentTerminalStore().balanceOf(_v3Terminal, _allocationDestinationProjectId) < 1 ether
```

Comment:
This pass if the balance is 0. Rather set the fee to 0 (terminal.setFee) or compute the net amount